src/presentation/rest/user/user.controller.ts

```diff
@@ -0,0 +1,182 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  Param,
+  Patch,
+  Post,
+  Query,
+} from '@nestjs/common';
+import { UserService } from 'src/application/user/user.service';
+import { FriendService } from '../../../application/friend/friend.service';
+import { CreateUserDto, UpdateUserDto } from './user.types';
+import { GeoPosition } from '../../../domain/address.type';
+import { User } from 'src/domain/user/user.types';
+import { ApiQuery } from '@nestjs/swagger';
+
+@Controller('user')
+export class UserController {
+  constructor(
+    private readonly userService: UserService,
+    private readonly friendService: FriendService,
+  ) {}
+
+  @Post()
+  create(@Body() createUserDto: CreateUserDto) {
+    return this.userService
+      .create({
+        ...createUserDto,
+        address: this.transformLatLongToPostgresGeoPosition(
+          createUserDto.address,
+        ),
+      })
+      .then((res) => this.mapUserAddressToOutside(res));
```

Comment:
Isn't it better to use pipes / interceptors for those transformations?

Reply:
Indeed, thanks for the heads up. Gonna rewrite that logic using interceptors.

Reply:
Thanks, just a reminder it's not necessary to put a fix or improvement for those comments.
After all, it's a test, we are not expecting a product-ready code here. :P

Reply:
Fair enough, I just took the opportunity to finish it :)